lib/Auth/Process/AttributeMap.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\Module\perun\Auth\Process;
+
+use SimpleSAML\Configuration;
+use SimpleSAML\Error\Exception;
+use SimpleSAML\Logger;
+use SimpleSAML\Module\perun\Adapter;
+
+class AttributeMap extends \SimpleSAML\Auth\ProcessingFilter
+{
+    public const MAP_ATTR_NAME = 'attrMapAttr';
+    public const KEEP_SOURCE_ATTRIBUTES = 'keepSourceAttributes';
+    public const ENTITY_ID = 'entityid';
+    public const INTERFACE_PROPNAME = 'interface';
+
+    public const CLASS_PREFIX = 'perun:AttributeMap: ';
+    public const RPC = 'rpc';
+    public const SP_METADATA = 'SPMetadata';
+    public const ATTRIBUTES = 'Attributes';
+
+    private $mapAttrName;
+    private $keepSourceAttributes;
+    private $entityId;
+    private $adapter;
+
+    public function __construct($config, $reserved)
+    {
+        parent::__construct($config, $reserved);
+        $config = Configuration::loadFromArray($config);
+
+        $this->mapAttrName = $config->getString(self::MAP_ATTR_NAME);
+        if (null === $this->mapAttrName) {
+            throw new Exception(self::CLASS_PREFIX . 'Missing mandatory configuration option \'mapAttrName\'');
+        }
+
+        $this->keepSourceAttributes = $config->getBoolean(self::KEEP_SOURCE_ATTRIBUTES, false);
+        $this->entityId = $config->getValue(self::ENTITY_ID);
+
+        $interface = $config->getString(self::INTERFACE_PROPNAME, self::RPC);
+        $this->adapter = Adapter::getInstance($interface);
+    }
+
+    public function process(&$request)
+    {
+        if (null === $this->entityId) {
+            $this->entityId = $request[self::SP_METADATA][self::ENTITY_ID];
+        } elseif (is_callable($this->entityId)) {
+            $this->entityId = call_user_func($this->entityId, $request);
+        } elseif (!is_string($this->entityId)) {
+            throw new Exception(self::CLASS_PREFIX . 'entityid must be a string or a callable');
+        }
+
+        $facility = $this->adapter->getFacilityByEntityId($this->entityId);
+
+        if (null === $facility) {
+            Logger::info(
+                self::CLASS_PREFIX . 'Facility with entityid ' . $this->entityId . ' not found. Skipping the filter'
+            );
+
+            return;
+        }
+
+        $attrMap = $this->adapter->getFacilityAttributesValues($facility, [$this->mapAttrName])[self::MAP_ATTR_NAME];
+        $requestAttributes = &$request[self::ATTRIBUTES];
+
+        foreach ($attrMap as $key => $value) {
```

Comment:
```suggestion
        foreach ($attrMap as $targetAttribute => $sourceAttribute) {
```
reversal of the map allows mapping the same source attribute into multiple target attributes